lib/Server.php

```diff
@@ -245,7 +242,18 @@ private function doStart(callable $bindSockets): \Generator {
 
         $this->vhosts->setupHttpDrivers($this->createHttpDriverHandlers(), $this->callableFromInstanceMethod("writeResponse"));
 
-        $this->boundServers = yield $bindSockets($this->generateBindableAddressContextMap());
+        $socketBinder = function ($address, $context) {
+            if (!strncmp($address, "unix://", 7)) {
+                @unlink(substr($address, 7));
+            }
+
+            if (!$socket = stream_socket_server($address, $errno, $errstr, STREAM_SERVER_BIND | STREAM_SERVER_LISTEN, stream_context_create($context))) {
+                throw new \RuntimeException(sprintf("Failed binding socket on %s: [Err# %s] %s", $address, $errno, $errstr));
+            }
+
+            return $socket;
+        };
+        $this->boundServers = yield $bindSockets($this->generateBindableAddressContextMap(), $socketBinder);
```

Comment:
This should probably be `$this->boundSockets`?

Reply:
if you want to be precise you could call it boundServerSockets - but really doesn't matter. I just retained the existing name.